core/domain/suggestion_services.py

```diff
@@ -1826,3 +1828,896 @@ def update_question_suggestion(
+    user and converts them to their corresponding domain objects.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        list(QuestionContributionStats). QuestionContributionStats domain
+        objects corresponding to the supplied user.
+    """
+    question_contribution_stats_models = (
+        suggestion_models.QuestionContributionStatsModel.get_all_by_user_id(
+            user_id
+        )
+    )
+    return [
+        _create_question_contribution_stats_from_model(model)
+        for model in question_contribution_stats_models
+    ]
+
+
+def _get_all_question_review_stats(
+    user_id: str
+) -> List[suggestion_registry.QuestionReviewStats]:
+    """Gets all QuestionReviewStatsModels corresponding to the supplied
+    user and converts them to their corresponding domain objects.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        list(QuestionReviewStats). QuestionReviewStats domain objects
+        corresponding to the supplied user.
+    """
+    question_review_stats_models = (
+        suggestion_models.QuestionReviewStatsModel.get_all_by_user_id(
+            user_id
+        )
+    )
+    return [
+        _create_question_review_stats_from_model(model)
+        for model in question_review_stats_models
+    ]
+
+
+def get_all_contributor_stats(
+    user_id: str
+) -> suggestion_registry.ContributorStatsSummary:
+    """Gets ContributorStatsSummary corresponding to the supplied user.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        ContributorStatsSummary. ContributorStatsSummary domain objects
+        corresponding to the supplied user.
+    """
+    translation_contribution_stats = get_all_translation_contribution_stats(
+        user_id)
+    translation_review_stats = _get_all_translation_review_stats(user_id)
+    question_contribution_stats = _get_all_question_contribution_stats(user_id)
+    question_review_stats = _get_all_question_review_stats(user_id)
+
+    return suggestion_registry.ContributorStatsSummary(
+        user_id,
+        translation_contribution_stats,
+        question_contribution_stats,
+        translation_review_stats,
+        question_review_stats)
+
+
+def _update_translation_contribution_stats_models(
+    translation_contribution_stats: List[
+        suggestion_registry.TranslationContributionStats]
+) -> None:
+    """Updates TranslationContributionStatsModel models for given translation
+    contribution stats.
+
+    Args:
+        translation_contribution_stats: list(TranslationContributionStats).
+            A list of TranslationContributionStats domain objects.
+    """
+    stats_dict = {}
+
+    for stat in translation_contribution_stats:
+        stat_id = (
+            suggestion_models.TranslationContributionStatsModel.construct_id(
+                str(stat.language_code),
+                str(stat.contributor_user_id),
+                str(stat.topic_id))
```

Comment:
As it is mentioned [here](https://github.com/oppia/oppia/pull/15917#discussion_r962264396) `TranslationContributionStatsModel.construct_id` expects a strings but these attributes in `suggestion_registry.TranslationContributionStats` are `Optional[str]`. It is obvious that these attributes are not None for this case since `TranslationContributionStats` objects passed in to `_update_translation_contribution_stats_models` will contain string values for these attributes. Anyway I added an exception when they are None.

Reply:
I'm leaving the resolution of this to @sahiljoster32 as it relates to his comments.

Reply:
Shouldn't TranslationContributionStats be typed in such a way that these attributes are not optional, in that case (even though the superclass attrs might be optional)?

/cc @sahiljoster32

Reply:
Yeah every translation contribution stat should have a language_code, contributor_user_id and a topic_id.